asyncscope.md

```diff
@@ -359,39 +370,35 @@ Examples of use
 
 ## Spawning work from within a task
 
-Use a `counting_scope` in combination with a `system_context` from [@P2079R2] to spawn work from within a task and join
-it later:
+Use a `let_with_async_scope` in combination with a `system_context` from [@P2079R2] to spawn work from within a task:
 ```cpp
 namespace ex = std::execution;
 
 int main() {
     ex::system_context ctx;
     int result = 0;
-    ex::counting_scope scope;
 
     ex::scheduler auto sch = ctx.scheduler();
 
-    ex::sender auto val = ex::on(sch, ex::just() | ex::then([&result, sch, &scope]() {
+    ex::sender auto val = ex::on(sch, ex::just() | ex::let_with_async_scope([sch](auto scope) {
         int val = 13;
 
         auto print_sender = ex::just() | ex::then([val] {
             std::cout << "Hello world! Have an int with value: " << val << "\n";
         });
-
+        
         // spawn the print sender on sch to make sure it completes before shutdown
         ex::spawn(scope, ex::on(sch, std::move(print_sender)));
 
-        return val;
+        return ex::just(val);
     })) | ex::then([&result](auto val) { result = val });
 
-    ex::spawn(scope, std::move(val));
-
-    this_thread::sync_wait(scope.join());
+    this_thread::sync_wait(std::move(val));
 
     std::cout << "Result: " << result << "\n";
 }
 
-// The counting scope ensured that all work is safely joined, so result contains 13
+// 'let_with_async_scope' ensured that all work is completed, so result contains 13
```

Comment:
Should we update the wording to address the possibility that the `spawn` on line 391 threw? Maybe mention that the `sync_wait` will throw whatever is thrown by the callable passed to `let_with_async_scope`?